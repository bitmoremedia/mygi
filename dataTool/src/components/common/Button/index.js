import React, { PropTypes } from 'react'

import { Button, DangerButton } from './styles'

const ButtonComponent = ({ children, onClick, danger }) => {
  if (danger) {
    return (
      <DangerButton onClick={onClick} >
        {children}
      </DangerButton>
    )
  }
  return (
    <Button onClick={onClick} >
      {children}
    </Button>
  )
}

ButtonComponent.propTypes = {
  children: PropTypes.any.isRequired,
  onClick: PropTypes.func,
  danger: PropTypes.bool,
}

ButtonComponent.defaultProps = {
  onClick: undefined,
  danger: false,
}

export default ButtonComponent
