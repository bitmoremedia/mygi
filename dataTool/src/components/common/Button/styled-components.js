import styled from 'styled-components'

export const Button = styled.button`
  background-color: black;
  color: white;
  cursor: pointer;
  border-radius: 1px;
  border: 0px;
  padding: 5px 10px 5px 10px;
  font-size: 14px;
  &:focus {
      outline:none;
  }
`

export const DangerButton = styled(Button)`
  background-color: red;
`
