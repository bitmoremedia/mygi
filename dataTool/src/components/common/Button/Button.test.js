import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'

import Button from './index'

test('component renders without error', () => {
  const wrapper = mount(
    <Button>
      Button Label
    </Button>,
  )
  expect(wrapper.exists()).toBe(true)
})

// INTERACTIVITY TESTS

test('onclick function is fired', () => {
  const onClick = jest.fn()
  const wrapper = mount(
    <Button onClick={onClick}>
      Button Label
    </Button>,
  )
  expect(onClick).not.toHaveBeenCalled()
  wrapper.find('button').simulate('click')
  expect(onClick).toHaveBeenCalled()
})

// SNAPSHOT TESTS

test('component renders as expected', () => {
  const component = renderer.create(
    <Button>
      Button Label
    </Button>,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('component renders a dangerous button', () => {
  const component = renderer.create(
    <Button danger>
      Button Label
    </Button>,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
