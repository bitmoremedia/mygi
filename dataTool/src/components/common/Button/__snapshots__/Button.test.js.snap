exports[`test component renders a dangerous button 1`] = `
<button
  className="feUtjL"
  onClick={undefined}>
  Button Label
</button>
`;

exports[`test component renders as expected 1`] = `
<button
  className="OgMJe"
  onClick={undefined}>
  Button Label
</button>
`;
